refactor: 사용자 정보 조회 로직 통합

devjaeyoon opened the pull request:
### ✨ 이슈 번호

- #68

### 📌 설명

현재 사용자의 즐겨찾기 채널 목록 정보는 Home 컴포넌트에 진입했을 때, 설정 정보는 Settings 컴포넌트에 진입했을 때 가져오고 있습니다. 우리의 서비스는 ChannelPlayer 컴포넌트에서 설정을 변경할 수 있고 사용자가 Home 컴포넌트에서 Settings 컴포넌트에 진입하지 않고 ChannelPlayer 컴포넌트에 진입하면 설정을 불러오지 않을 수 있는 에러가 발생할 수 있습니다. 이를 사전에 방지하기 위해 메인 페이지인 Home 컴포넌트에 진입할 때 사용자의 정보를 모두 받아올 수 있도록 사용자 정보 조회 로직 통합 작업을 진행한 Pull Request입니다.

### 📃 작업 사항

- [x] 사용자 정보 조회 로직 통합
- [x] 통합하며 제거해도 되는 코드들 제거

### 💡 참고 사항

useUserProfile은 기존에는 useUserProfile 훅 내에서 localStorage에서 직접 userId를 가져왔으나, 이를 외부에서 주입받도록 변경하여 훅의 재사용을 개선했습니다.

### 💭 리뷰 사항 반영

- [x] `useUserData`에서 `useChannels` 제거

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

devjaeyoon committed:
refactor: useUserProfile 훅에서 userId를 인자로 받도록 수정

기존에는 useUserProfile 훅 내에서 localStorage에서 직접 userId를 가져왔으나,
이를 외부에서 주입받도록 변경하여 훅의 재사용성을 개선함.
useUserData 훅과 통합 구조에 적합하도록 구조 리팩터링 목적의 변경.

devjaeyoon committed:
feat: useUserData 훅 생성 및 사용자 초기 데이터 로딩 통합

사용자의 채널 목록, 관심 채널, 설정 정보를 한 번에 가져오는 커스텀 훅 생성.
기존 페이지별 분산된 호출 로직을 통합하여 코드 구조 일관성 향상 및 재사용성 확보.

devjaeyoon committed:
refactor: 사용자 초기 데이터 로딩을 Home에서 일괄 처리하도록 구조 통합

Home 컴포넌트에 useUserData 훅을 도입해 사용자 관련 데이터(채널, 관심 채널, 설정)를 통합 로딩.
Settings 컴포넌트에서 중복되던 useUserProfile 호출 제거하여 책임 명확화 및 중복 방지.

faber-h commented on src/hooks/useUserData.jsx:
```diff
@@ -0,0 +1,14 @@
+const useUserData = () => {
+  const userId = useUserId();
+
+  useChannels();
```
`user`와 관련된 데이터를 잘 분리해주셨네요!
다만, `useChannels`는 전체 채널을 불러오는 역할을 하고 있기 때문에, `useUserData`와는 관심사가 다르다고 생각해요.
`useUserData` 내부에서 `useChannels`를 호출하지 않는 방향이 관심사의 분리에 도움이 될 것 같아요 :)

devjaeyoon replied:
맞습니다. `useChannels`는 전체 채널을 불러오는 로직이라 사용자의 데이터를 가져오는 `useUserData`와는 상응하지 않네요. 해당 내용 수정하겠습니다!

devjaeyoon committed:
refactor: useChannels 훅을 useUserData에서 분리하여 관심사 분리 강화

useChannels는 전체 채널 데이터를 다루는 역할로, 사용자 관련 데이터를 처리하는
useUserData와 분리하는 것이 적절하다고 판단. 관심사 분리를 통해
각 훅의 목적을 명확히 하고 유지보수성을 향상시킴.

Eun0713 commented on src/hooks/useUserProfile.jsx:
```diff
@@ -3,12 +3,11 @@ import { useEffect } from "react";
+const useUserProfile = (userId) => {
   const { setUserSettings } = useUserStore();
 
   useEffect(() => {
```
```suggestion
  useEffect(() => {
  if (!userId) {
    return;
  }
```
`useEffect`진입 초반에 early return으로 처리하면 함수 선언을 생략할 수 있어 코드가 조금 더 간결하고 의도가 명확해질 것 같습니다 :) ❕

d0-neii commented on src/hooks/useUserProfile.jsx:
```diff
@@ -3,12 +3,11 @@ import { useEffect } from "react";
+const useUserProfile = (userId) => {
   const { setUserSettings } = useUserStore();
 
   useEffect(() => {
```
```suggestion
  useEffect(() => {
    if (!userId) {
      return;
    }
```
`useEffect`진입 초반에 early return으로 처리하면 함수 선언을 생략할 수 있어 코드가 조금 더 간결하고 의도가 명확해질 것 같습니다 :) ❕

devjaeyoon replied:
불필요한 함수 생성을 막을 수 있는 점이 특히 좋은 것 같아요! 좋은 리뷰 감사드립니다! 해당 리뷰 반영했습니다!

devjaeyoon committed:
refactor: userId가 없을 경우 조기 반환하여 불필요한 함수 생성 방지

useEffect 내부에서 userId 유효성 검사를 먼저 수행하여,
initUserProfile 함수 생성을 조건적으로 하지 않도록 개선.

Eun0713 reviewed (approved):
LGTM🔥

faber-h reviewed (approved):
LGTM 🔥
